### PR TITLE
Filter scalar reserve requirements

### PR DESCRIPTION
**Pull request by darrylmelander**
Use the Reserve_Products row of simulation_objects.csv to decide which scalar reserve products to included in a model.

Note that this only applies to scalar reserve requirements (the ones listed in reserves.csv). It does not filter timeseries by reserve type; if a timeseries is listed in timeseries_pointers.csv for a given reserve, the timeseries is included in the model
even if the reserve type isn't listed in simulation_objects.csv.

If simulation_objects.csv does not have a Reserve_Products row then all scalar reserve products are accepted for both DAY_AHEAD and REAL_TIME models.

To summarize:
* If a reserve product has a timeseries for the relevant simulation type, the timeseries will be included.
* If a reserve product does not have a timeseries but does have a row in reserves.csv, the reserve will be specified with a scalar value if one of the following is true:
    * The reserve product is listed in simulation_objects.csv in the Reserve_Products row under the relevant simulation type, OR
    * There is no ReserveProducts row in simulation_objects.csv.

### CODE REVIEW

**Review by bknueven — APPROVED** [review]
Code looks good; behaves as-expected on RTS-GMLC data.
